src/common/set.c

```diff
@@ -2084,6 +2084,25 @@ util_poolset_create_set(struct pool_set **setp, const char *path,
 	return ret;
 }
 
+/*
+ * util_poolset_check_options -- (internal) check if poolset options match given
+ *                               flags
+ */
+static int
+util_poolset_check_options(struct pool_set *set, uint32_t incompat)
```

Comment:
and now out comes the dichotomy of the poolset config. Maybe OPTION_NO_HDRS should be a create-time option instead of a poolset option (like I initially suggested)?